content/content-tool/tool/src/java/org/sakaiproject/content/tool/FilePickerAction.java

```diff
@@ -1359,6 +1360,9 @@ else if(fileContentStream != null)
 				// make an attachment resource for this URL
 				try
 				{
+					if (!Validator.checkFilename(filename)){
+						throw new InvalidFilenameException(filename);
```

Comment:
Same again, I don't like this pattern, throwing stuff which is just caught a bit further on in the same method. Just nest the code into a block.

Reply:
Yeah, agree, the pattern is kinda bad, going to replace it.